### PR TITLE
PSMDB-1212 moderinize an cleanup of cyrus_sasl_server_session.*

### PR DESCRIPTION
**Pull request by igorsol**
Most changes are style improvements or just formatting changes.
The most interesting part is the remove of deprecated `throw()` construct. In fact it could cause termination of our process because exceptions are possible inside `saslSessionProxyPolicy`

### CODE REVIEW

File: src/mongo/db/auth/external/cyrus_sasl_server_session.cpp [hunk 1]
```diff
@@ -81,15 +81,16 @@ struct gss_result {
...
             throw std::runtime_error(fmt::format("{} error: major: {}; minor: {}",
                                                  loc, major, minor));
+        }
     }
 };
 
 class auto_gss_name_t {
 public:
-    auto_gss_name_t() : _v(nullptr) {}
+    auto_gss_name_t() = default;
```
Comment (ktrushin) [inline src/mongo/db/auth/external/cyrus_sasl_server_session.cpp:93]: Can we remove the default constructor declaration/definition and rely on the compiler automatically generating one?

File: src/mongo/db/auth/external/cyrus_sasl_server_session.cpp [hunk 2]
```diff
@@ -238,7 +241,7 @@ StatusWith<std::tuple<bool, std::string> > CyrusSASLServerSession::step(StringDa
 }
 
 StringData CyrusSASLServerSession::getPrincipalName() const {
-    const char* username;
+    const char* username = nullptr;
     int result = sasl_getprop(_saslConnection, SASL_USERNAME, (const void**)&username);
```
Comment (ktrushin) [inline src/mongo/db/auth/external/cyrus_sasl_server_session.cpp:245]: In that [line](https://github.com/percona/percona-server-mongodb/pull/1001/files#diff-76b5162fc32296d3e51c36921e7d2ddc92bd8fed72375132ff020e6153d6e5a4R116), the C-style conversion was replaced with a `static_cast`. Is it worth using `static_cast`/`reinterpret_cast` in the other places in the file where old-fashioned conversions are still used?

**Review by ktrushin — CHANGES_REQUESTED** [review]
Since the purpose of the patch is improving style and formatting, I suggest not limiting ourselves to what has been already done but going further and passing these two files in their entirety through `clang-format`. That should fix a couple of other minor issues. E.g. in this [line](https://github.com/percona/percona-server-mongodb/pull/1001/files#diff-76b5162fc32296d3e51c36921e7d2ddc92bd8fed72375132ff020e6153d6e5a4R177), formatting seems unintuitive: `canon_auth_identity` is an argument of `fmt::format` but aligned with `saslSetError`. That [line](https://github.com/percona/percona-server-mongodb/pull/1001/files#diff-76b5162fc32296d3e51c36921e7d2ddc92bd8fed72375132ff020e6153d6e5a4R211) exceeds width limit, which is 100 as per the `.clang-format` file.

File: src/mongo/db/auth/external/cyrus_sasl_server_session.cpp (hunk 2 above)
Comment (igorsol) [reply src/mongo/db/auth/external/cyrus_sasl_server_session.cpp]: I updated PR. Unfortunately no way to use static_cast's. Due to ugly C libraries API I had to use const_cast and reinterpret_cast.

**Review by ktrushin — APPROVED** [review]
Look good

**Pull request merged**